agent/engine/execcmd/agent_version.go

```diff
@@ -15,68 +15,107 @@ package execcmd
 
 import (
 	"errors"
+	"io/ioutil"
 	"strconv"
 	"strings"
 )
 
-// TODO: Remove this later, just doing it do make static checks happy about these utils not being invoked anywhere
-func init() {
-	determineLatestVersion([]string{"1.1.1.1"})
-}
-
 const (
 	versionDelimiter  = "."
-	badInputReturn    = -999
 	badFormatErrorStr = "invalid version format"
 )
 
-// compareAgentVersion compares two versions, returning -1 if v1 < v2, 0 if v1 == v2, and 1 if v1 > v2.
-// The version must have only integers representing major, minor, patch, and reserved; delimited by dots (e.g. "1.2.3.4").
-// An error is returned if any of the versions passed as parameter is malformed.
-// example:
-// compareAgentVersion("3.0.236.0", "3.0.237.0") returns -1, nil
-// compareAgentVersion("3.0.236.0", "3.0.236.0") returns 0, nil
-// compareAgentVersion("3.0.237.0", "3.0.236.0") returns 1, nil
-// compareAgentVersion("a.b.c.d", "3.0.236.0") returns -999, error
-func compareAgentVersion(v1Str, v2Str string) (int, error) {
-	v1Arr := strings.Split(v1Str, versionDelimiter)
-	v2Arr := strings.Split(v2Str, versionDelimiter)
+type agentVersion []uint
```

Comment:
I think this could be `[4]uint` instead of slices

Reply:
or use struct for this

```go
type AssemblyVersion struct {
    Major uint
    Minor uint
    Build uint
    Revision uint
}
```

, but since this type is very lightly used, don't this is necessary for now

Reply:
why limit this and make it too rigid? since in the future the version could be less numbers or more (we don't really control that). Right now this works with whatever number of elements you have in the version

Reply:
ok